operator-common/src/main/java/io/strimzi/operator/cluster/model/Ca.java

```diff
@@ -100,6 +101,8 @@
 
     private final PasswordGenerator passwordGenerator;
     protected final Reconciliation reconciliation;
+    private Clock clock;
+    private static final ZoneId UTC = ZoneId.of("UTC");
```

Comment:
Actually this is how the `OpenSslCertManager` works. It uses the same logic and when a certificate is generated it uses UTC time for expiration (notBefore, notAfter). See [here](https://github.com/strimzi/strimzi-kafka-operator/blob/main/certificate-manager/src/main/java/io/strimzi/certs/OpenSslCertManager.java#L143).
In our `Ca`, the logic to compare current time with expiration and then removing certificate is currently using the `Instant.now()` which uses a UTC system clock by default.
The addition of a `Clock` instance in the `Ca` is purely for testing. By default it will use the UTC system clock having the same behaviour as today but it's useful to be changed as demonstrated in my added test.

Reply:
We don't seem to actually use the zone.

Reply:
Yeah that's true even because the default `Clock` is UTC based. I mislead Jakub question then :-)